README.md
# DesdeCasa (StayAtHome)

![Continuous deployment](https://github.com/ferlopezm94/desdecasa/workflows/Continuous%20deployment/badge.svg?branch=master)

DesdeCasa is a website that displays the most relevant information (daily and historical) related to COVID-19 in Mexico, both national and statal.

The objective is to have an easy to use website for people to share the information and raise awareness about the magnitude of the problem.

## Data

### Input

The data shared by [The Health Department](https://www.gob.mx/salud) of Mexico's government is used to display the information in the website. It's published as Open Data and you can find it [here](https://datos.gob.mx/busca/dataset/informacion-referente-a-casos-covid-19-en-mexico).

The data is downloaded and processed. You can find both formats [here](https://github.com/ferlopezm94/desdecasa/tree/master/src/data).

### Output

The information shown in the website is the following:

- Confirmed cases
- Deaths
- Suspect cases
- Active cases
- Tests made

All but tests made are shared for both national and statal level. The information is updated daily at around 20:00 hrs and the users can view the information in a daily or historical basis using some charts.

## Getting started

### Requirements

Make sure the following tools are installed in your system:

- [Node v12.X.X](https://nodejs.org/en/download/)
- [Yarn v1.21.X](https://yarnpkg.com/en/docs/install)

### Installation

Clone the GitHub repository and user `yarn` to install the dependencies.

```
$ git clone https://github.com/ferlopezm94/desdecasa.git
$ cd desdecasa
$ yarn install
```

## Development

### Local

```
$ yarn develop
```

Now you're ready to view the website, open a browser and visit `localhost:8000`. You can start making some changes and each one will trigger an update to the website.

### Build

To create a production build simply run:

```
$ yarn build
```

Gatsby will perform an optimized production build for your site, generating static HTML and per-route JavaScript code bundles. The build will be in the `public` folder.

To serve the production build locally run:

```
$ yarn serve
```

Gatsby starts a local HTML server for testing your build site at localhost:9000.

## Directory layout

```
.
├── public/                 <-- Static files optimized for a production build
├── scripts/                <-- Logic to parse the raw COVID-19 data and create the JSON formats used in the website
├── source/                 <-- Source code for our website
```

## Contributions

Open an issue or add a Bug or Feature request, or send me a DM in [twitter](https://twitter.com/ferlopezm94) and we can talk.


CHANGELOG.md
## [1.13.3](https://github.com/ferlopezm94/desdecasa/compare/v1.13.2...v1.13.3) (2020-11-17)


### Bug Fixes

* **ci:** replace set-env command ([4c3c4eb](https://github.com/ferlopezm94/desdecasa/commit/4c3c4eb626606d0b280a84b89f712a99be07e64a))

## [1.13.2](https://github.com/ferlopezm94/desdecasa/compare/v1.13.1...v1.13.2) (2020-06-07)


### Bug Fixes

* **data:** update 2020-06-06 ([7b2f5ef](https://github.com/ferlopezm94/desdecasa/commit/7b2f5efa70c8efb9af1cb21aeb756556256f406f))

## [1.13.1](https://github.com/ferlopezm94/desdecasa/compare/v1.13.0...v1.13.1) (2020-06-06)


### Bug Fixes

* **data:** update 2020-06-05 ([9dd80ca](https://github.com/ferlopezm94/desdecasa/commit/9dd80ca08cbe5a35e4be3987992bf115861e0be0))

# [1.13.0](https://github.com/ferlopezm94/desdecasa/compare/v1.12.0...v1.13.0) (2020-05-22)


### Features

* **data:** add function to retrieve and parse zip file ([7b28187](https://github.com/ferlopezm94/desdecasa/commit/7b2818719c960cd70716a0df0fccb9d0de2278bf))

# [1.12.0](https://github.com/ferlopezm94/desdecasa/compare/v1.11.4...v1.12.0) (2020-04-27)


### Bug Fixes

* **top:** rename states column ([9f46d15](https://github.com/ferlopezm94/desdecasa/commit/9f46d158838e17ba358f4035cc2e0ee5a949f090))


### Features

* **sharing:** add amplitude events and whatsapp sharing ([ca765f8](https://github.com/ferlopezm94/desdecasa/commit/ca765f895abf4e461ba996fe30cd12d428ee9c40))

## [1.11.4](https://github.com/ferlopezm94/desdecasa/compare/v1.11.3...v1.11.4) (2020-04-26)


### Bug Fixes

* **top:** add offset in x axis of players ([0320e03](https://github.com/ferlopezm94/desdecasa/commit/0320e032c0731c727f694d63751d274c8ba094aa))

## [1.11.3](https://github.com/ferlopezm94/desdecasa/compare/v1.11.2...v1.11.3) (2020-04-26)


### Bug Fixes

* **race:** change background ([e8ba72d](https://github.com/ferlopezm94/desdecasa/commit/e8ba72d065e8fb10268501b964247dd1eb907f6d))
* **top:** change messaging ([c30adc9](https://github.com/ferlopezm94/desdecasa/commit/c30adc99a92e3dd98390c4abeb9eb9579c67b901))
* **top:** update seo component path ([721675f](https://github.com/ferlopezm94/desdecasa/commit/721675fae97ea061602d4d6b79b6e15eac87d7bd))

## [1.11.2](https://github.com/ferlopezm94/desdecasa/compare/v1.11.1...v1.11.2) (2020-04-26)


### Bug Fixes

* **race:** change bottom position in players ([8f349c8](https://github.com/ferlopezm94/desdecasa/commit/8f349c802ffb5f7a43cb71d9f834a935d66b01a0))

## [1.11.1](https://github.com/ferlopezm94/desdecasa/compare/v1.11.0...v1.11.1) (2020-04-26)


### Bug Fixes

* **race:** change bottom position in players ([a618980](https://github.com/ferlopezm94/desdecasa/commit/a618980242679f21bedfaed260aaefca1320ed9f))

# [1.11.0](https://github.com/ferlopezm94/desdecasa/compare/v1.10.1...v1.11.0) (2020-04-26)


### Features

* **race:** add custom gif ([c2ff2be](https://github.com/ferlopezm94/desdecasa/commit/c2ff2be479f7b271c7e18b09a2f5da1bf10752b5))
* **race:** add dashboard with top ten list ([a9aef8a](https://github.com/ferlopezm94/desdecasa/commit/a9aef8ac71c38d4aaa9455a061640a8206669c1c))
* **race:** add sliding background ([b896086](https://github.com/ferlopezm94/desdecasa/commit/b896086000d063e1acb201a805206183f846ea0c))
* **race:** add state name ([9fbb7d7](https://github.com/ferlopezm94/desdecasa/commit/9fbb7d71534428a2de0b367673da991116cc0778))
* **race:** create reusable player ([17f0d9f](https://github.com/ferlopezm94/desdecasa/commit/17f0d9f369f4c4962a4a6846f929f1e2b7e4b038))
* **race:** display top ten players ([c44ea6d](https://github.com/ferlopezm94/desdecasa/commit/c44ea6d80685838b9f8a0baedd84e14fca1d9033))

## [1.10.1](https://github.com/ferlopezm94/desdecasa/compare/v1.10.0...v1.10.1) (2020-04-26)


### Bug Fixes

* **index:** only change to state view when complete visibility of map ([ffda220](https://github.com/ferlopezm94/desdecasa/commit/ffda22079a107d15856684eef583ed5adf2f9603))

# [1.10.0](https://github.com/ferlopezm94/desdecasa/compare/v1.9.0...v1.10.0) (2020-04-25)


### Bug Fixes

* **index:** increase z-index in basic mode control ([dff6f3a](https://github.com/ferlopezm94/desdecasa/commit/dff6f3ab43d298000a031761fb7da9f71df06ad4))


### Features

* **charts:** add new national tests by day ([3619d38](https://github.com/ferlopezm94/desdecasa/commit/3619d38b9b6525d2b9cca0975ea0afffa08b5798))
* **charts:** show new daily confirmed and deaths instead of accumulated ([4d8ec63](https://github.com/ferlopezm94/desdecasa/commit/4d8ec63c42713188c8adc37fefeb039eff0eb21e))
* **scripts:** add total national tests for historical data ([d5dd9ed](https://github.com/ferlopezm94/desdecasa/commit/d5dd9ed556464536f1919e2c4cf9c9fcabe99610))

# [1.9.0](https://github.com/ferlopezm94/desdecasa/compare/v1.8.0...v1.9.0) (2020-04-25)


### Features

* **index:** add national tests stat ([4e8576e](https://github.com/ferlopezm94/desdecasa/commit/4e8576ebf027a083446441289391c5f886cbfb17))

# [1.8.0](https://github.com/ferlopezm94/desdecasa/compare/v1.7.0...v1.8.0) (2020-04-25)


### Bug Fixes

* **data:** update script for creting daily data by state ([5310c40](https://github.com/ferlopezm94/desdecasa/commit/5310c408552f8b52cc6d9a15236783b7bafc7b33))


### Features

* **data:** add new parsing data script and information of 2020-04-20 ([80c0044](https://github.com/ferlopezm94/desdecasa/commit/80c00442abcfa5eb41496c3e6aaf021864d64a99))

# [1.7.0](https://github.com/ferlopezm94/desdecasa/compare/v1.6.2...v1.7.0) (2020-04-19)


### Bug Fixes

* **data:** scroll to correct section when switching basic mode ([28d3d87](https://github.com/ferlopezm94/desdecasa/commit/28d3d874a77a5e2e6a01cf06d85d2bcad0a023f7))
* **state:** remove specific state page ([954e652](https://github.com/ferlopezm94/desdecasa/commit/954e65259611078c8bc8978040d3f32a6db5baeb))


### Features

* **charts:** show only 3 labels in x axis ([f3c15ff](https://github.com/ferlopezm94/desdecasa/commit/f3c15ff8e522330c75f7fed40713892ea65e39b5))

scripts/parse/utils.js
// const parseOrigen = (input) => {
//   let output;

//   switch (input) {
//     default:
//       output = 'DESCONOCIDO';
//       break;
//   }

//   return output;
// };

const expectedColumnsAndOrder = [
  'FECHA_ACTUALIZACION',
  'ID_REGISTRO',
  'ORIGEN',
  'SECTOR',
  'ENTIDAD_UM',
  'SEXO',
  'ENTIDAD_NAC',
  'ENTIDAD_RES',
  'MUNICIPIO_RES',
  'TIPO_PACIENTE',
  'FECHA_INGRESO',
  'FECHA_SINTOMAS',
  'FECHA_DEF',
  'INTUBADO',
  'NEUMONIA',
  'EDAD',
  'NACIONALIDAD',
  'EMBARAZO',
  'HABLA_LENGUA_INDIG',
  'INDIGENA',
  'DIABETES',
  'EPOC',
  'ASMA',
  'INMUSUPR',
  'HIPERTENSION',
  'OTRA_COM',
  'CARDIOVASCULAR',
  'OBESIDAD',
  'RENAL_CRONICA',
  'TABAQUISMO',
  'OTRO_CASO',
  'TOMA_MUESTRA',
  'RESULTADO_LAB',
  'CLASIFICACION_FINAL',
  'MIGRANTE',
  'PAIS_NACIONALIDAD',
  'PAIS_ORIGEN',
  'UCI',
];

const caseStateName = stateName => {
  let casedStateName;

  switch (stateName) {
    case 'COAHUILA DE ZARAGOZA':
      return 'Coahuila';
    case 'CIUDAD DE MÉXICO':
      return 'Ciudad de México';
    case 'MICHOACÁN DE OCAMPO':
      return 'Michoacán';
    case 'VERACRUZ DE IGNACIO DE LA LLAVE':
      return 'Veracruz';
    default:
      casedStateName = stateName.toLowerCase().split(' ');
      break;
  }

  for (let i = 0; i < casedStateName.length; i++) {
    casedStateName[i] = casedStateName[i].charAt(0).toUpperCase() + casedStateName[i].substring(1);
  }

  return casedStateName.join(' ');
};

const parseResultado = input => {
  let output;

  switch (input) {
    case 1:
      output = 'POSITIVO';
      break;
    case 2:
      output = 'NEGATIVO';
      break;
    case 3:
      output = 'PENDIENTE';
      break;
    default:
      output = 'DESCONOCIDO';
      break;
  }

  return output;
};

const parseMunicipio = input => {
  let output;

  switch (input) {
    default:
      output = 'DESCONOCIDO';
      break;
  }

  return output;
};

const parseSexo = input => {
  let output;

  switch (input) {
    case '1':
      output = 'HOMBRE';
      break;
    case '2':
      output = 'MUJER';
      break;
    default:
      output = 'DESCONOCIDO';
      break;
  }

  return output;
};

const parseEntidades = input => {
  let output;

  switch (input) {
    case 1:
      output = 'AGUASCALIENTES';
      break;
    case 2:
      output = 'BAJA CALIFORNIA';
      break;
    case 3:
      output = 'BAJA CALIFORNIA SUR';
      break;
    case 4:
      output = 'CAMPECHE';
      break;
    case 5:
      output = 'COAHUILA DE ZARAGOZA';
      break;
    case 6:
      output = 'COLIMA';
      break;
    case 7:
      output = 'CHIAPAS';
      break;
    case 8:
      output = 'CHIHUAHUA';
      break;
    case 9:
      output = 'CIUDAD DE MÉXICO';
      break;
    case 10:
      output = 'DURANGO';
      break;
    case 11:
      output = 'GUANAJUATO';
      break;
    case 12:
      output = 'GUERRERO';
      break;
    case 13:
      output = 'HIDALGO';
      break;
    case 14:
      output = 'JALISCO';
      break;
    case 15:
      output = 'MÉXICO';
      break;
    case 16:
      output = 'MICHOACÁN DE OCAMPO';
      break;
    case 17:
      output = 'MORELOS';
      break;
    case 18:
      output = 'NAYARIT';
      break;
    case 19:
      output = 'NUEVO LEÓN';
      break;
    case 20:
      output = 'OAXACA';
      break;
    case 21:
      output = 'PUEBLA';
      break;
    case 22:
      output = 'QUERÉTARO';
      break;
    case 23:
      output = 'QUINTANA ROO';
      break;
    case 24:
      output = 'SAN LUIS POTOSÍ';
      break;
    case 25:
      output = 'SINALOA';
      break;
    case 26:
      output = 'SONORA';
      break;
    case 27:
      output = 'TABASCO';
      break;
    case 28:
      output = 'TAMAULIPAS';
      break;
    case 29:
      output = 'TLAXCALA';
      break;
    case 30:
      output = 'VERACRUZ DE IGNACIO DE LA LLAVE';
      break;
    case 31:
      output = 'YUCATÁN';
      break;
    case 32:
      output = 'ZACATECAS';
      break;
    case 36:
      output = 'ESTADOS UNIDOS MEXICANOS';
      break;
    case 97:
      output = 'NO APLICA';
      break;
    case 98:
      output = 'SE IGNORA';
      break;
    case 99:
      output = 'NO ESPECIFICADO';
      break;
    default:
      output = 'DESCONOCIDO';
      break;
  }

  return output;
};

const parseSector = input => {
  let output;

  switch (input) {
    case '1':
      output = 'CRUZ ROJA';
      break;
    case '2':
      output = 'DIF';
      break;
    case '3':
      output = 'ESTATAL';
      break;
    case '4':
      output = 'IMSS';
      break;
    case '5':
      output = 'IMSS-BIENESTAR';
      break;
    case '6':
      output = 'ISSSTE';
      break;
    case '7':
      output = 'MUNICIPAL';
      break;
    case '8':
      output = 'PEMEX';
      break;
    case '9':
      output = 'PRIVADA';
      break;
    case '10':
      output = 'SEDENA';
      break;
    case '11':
      output = 'SEMAR';
      break;
    case '12':
      output = 'SSA';
      break;
    case '13':
      output = 'UNIVERSITARIO';
      break;
    case '99':
      output = 'NO ESPECIFICADO';
      break;
    default:
      output = 'DESCONOCIDO';
      break;
  }

  return output;
};

const parseOrigen = input => {
  let output;

  switch (input) {
    case '1':
      output = 'USMER';
      break;
    case '2':
      output = 'FUERA DE USMER';
      break;
    case '99':
      output = 'NO ESPECIFICADO';
      break;
    default:
      output = 'DESCONOCIDO';
      break;
  }

  return output;
};

module.exports = {
  caseStateName,
  expectedColumnsAndOrder,
  parseOrigen,
  parseSector,
  parseEntidades,
  parseSexo,
  parseMunicipio,
  parseResultado,
};
